File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/creation/tag/TagCreationEvent.kt
package dev.subfly.yabacore.state.creation.tag

import dev.subfly.yabacore.model.utils.YabaColor

sealed class TagCreationEvent {
    data class OnInitWithTag(val tagIdString: String?) : TagCreationEvent()
    data class OnSelectNewColor(val newColor: YabaColor) : TagCreationEvent()
    data class OnSelectNewIcon(val newIcon: String) : TagCreationEvent()
    data class OnChangeLabel(val newLabel: String) : TagCreationEvent()
    data class OnSave(
        val onSavedCallback: () -> Unit,
        val onErrorCallback: () -> Unit,
    ): TagCreationEvent()
}

File: Darwin/YABACore/State/Creation/Tag/TagCreationEvent.swift
//
//  TagCreationEvent.swift
//  YABACore
//
//  Parity with Compose `TagCreationEvent` — color as `YabaColor` (Compose `YabaColor.code`).
//

import Foundation

public enum TagCreationEvent: Sendable {
    case onInitWithTag(tagId: String?)
    case onSelectNewColor(YabaColor)
    case onSelectNewIcon(String)
    case onChangeLabel(String)
    case onSave
}
